src/utils/listener.js

```diff
@@ -22,21 +40,85 @@ Listener.prototype.destroy = function() {
     this._connection = null;
 };
 
+/*
+ * This function can be called by the provider within the callback function.
```

Comment:
This documentation is a bit vague, better to say something like

` 
Accepting a listener request informs deepstream that the current provider is willing
to provide the  record or event matching the subscriptionName . This will establish the current provider as the only publisher for the actual subscription with the deepstream cluster.
`